examples/permalink.js

```diff
@@ -98,6 +103,127 @@ app.MapDirectiveController = function(ngeoLocation, ngeoDebounce) {
 app.module.controller('AppMapController', app.MapDirectiveController);
 
 
+/**
+ * A draw directive that adds a simple draw tool.
+ *
+ * @return {angular.Directive} The directive specs.
+ */
+app.drawDirective = function() {
+  return {
+    restrict: 'E',
+    scope: {
+      'map': '=appDrawMap',
+      'layer': '=appDrawLayer'
+    },
+    controller: 'AppDrawController',
+    controllerAs: 'ctrl',
+    bindToController: true,
+    template:
+        '<label>Enable drawing:' +
+        '<input type="checkbox" ng-model="ctrl.interaction.active" />' +
+        '</label><br>' +
+        '<button ng-click="ctrl.clearLayer()">Clear layer</button>'
+
+  };
+};
+
+
+app.module.directive('appDraw', app.drawDirective);
+
+
+
+/**
+ * @param {angular.Scope} $scope Scope.
+ * @param {ngeo.DecorateInteraction} ngeoDecorateInteraction Decorate
+ *     interaction service.
+ * @param {ngeo.Location} ngeoLocation ngeo Location service.
+ * @constructor
+ * @export
+ * @ngInject
+ */
+app.DrawDirectiveController =
+    function($scope, ngeoDecorateInteraction, ngeoLocation) {
+
+  /**
+   * @type {ol.Map}
+   * @export
+   */
+  this.map;
+
+  /**
+   * @type {ol.layer.Vector}
+   */
+  this.layer;
+
+  /**
+   * @type {ngeo.Location}
+   * @private
+   */
+  this.ngeoLocation_ = ngeoLocation;
+
+  /**
+   * @type {number}
+   * @private
+   */
+  this.featureSeq_ = 0;
+
+  var vectorSource = this.layer.getSource();
+
+  /**
+   * @type {ol.interaction.Draw}
+   * @export
+   */
+  this.interaction = new ol.interaction.Draw(
+      /** @type {olx.interaction.DrawOptions} */ ({
+        type: 'LineString',
+        source: vectorSource
+      }));
```

Comment:
The `DrawOptions` record type has many more properties than what's used here. See https://github.com/openlayers/ol3/blob/master/externs/olx.js#L2370-L2386. This is why a cast is used here. But the compiler may now be able to infer the type. I'll try.